Vic2ToHoI4/Source/HOI4World/HoI4FocusTree.cpp

```diff
@@ -1906,65 +1989,589 @@ void HoI4FocusTree::addGPWarBranch(shared_ptr<HoI4::Country> Home, const vector<
+				newFocus->prerequisites.clear();
+				newFocus->prerequisites.push_back("= { focus = prepare_for_war" + tag + target.first + " }");
+				newFocus->relativePositionId += target.first;
+				newFocus->updateFocusElement(newFocus->available, "var:neighbor_war_defender", target.first);
+				newFocus->updateFocusElement(newFocus->available, "var:ROOT.neighbor_war_defender", target.first);
+				if (!majorIdeologies.count("communism"))
+				{
+					std::string comm = "";
+					comm += "\n\t\t\telse_if = {\n";
+					comm += "\t\t\t\tlimit = { var:neighbor_war_defender = { potential_communist_puppet_target = yes } }\n";
+					comm += "\t\t\t\tdeclare_war_on = {\n";
+					comm += "\t\t\t\t\ttarget = var:neighbor_war_defender\n";
+					comm += "\t\t\t\t\ttype = puppet_wargoal_focus\n";
+					comm += "\t\t\t\t}\n";
+					comm += "\t\t\t\tadd_named_threat = { threat = 3 name = neighbor_war_NF_threat }\n";
+					comm += "\t\t\t}";
+					newFocus->updateFocusElement(newFocus->completionReward, comm, "");
+				}
+				if (!majorIdeologies.count("fascism"))
+				{
+					std::string fasc = "";
+					fasc += "\n\t\t\telse_if = {\n";
+					fasc += "\t\t\t\tlimit = { var:neighbor_war_defender = { potential_fascist_annex_target = yes } }\n";
+					fasc += "\t\t\t\tdeclare_war_on = {\n";
+					fasc += "\t\t\t\t\ttarget = var:neighbor_war_defender\n";
+					fasc += "\t\t\t\t\ttype = annex_everything\n";
+					fasc += "\t\t\t\t}\n";
+					fasc += "\t\t\t\tadd_named_threat = { threat = 3 name = neighbor_war_NF_threat }\n";
+					fasc += "\t\t\t}";
+					newFocus->updateFocusElement(newFocus->completionReward, fasc, "");
+				}
+				newFocus->updateFocusElement(newFocus->completionReward, "var:neighbor_war_defender", target.first);
+				newFocus->updateFocusElement(newFocus->completionReward, "var:ROOT.neighbor_war_defender", target.first);
+				newFocus->updateFocusElement(newFocus->bypass, "var:neighbor_war_defender", target.first);
+				focuses.push_back(newFocus);
+			}
+			else
+			{
+				throw std::runtime_error("Could not load focus neighbor_war");
+			}
+
+			nextFreeColumn += 2;
+		}
+	}
+
+	return conquerTags;
+}
+
 void HoI4FocusTree::addNeighborWarBranch(
 	const string& tag,
-	const vector<shared_ptr<HoI4::Country>>& weakNeighbors,
+	const map<string, shared_ptr<HoI4::Country>>& closeNeighbors,
 	const shared_ptr<HoI4::Country>& targetNeighbors,
 	const string& targetName,
 	const date& startDate,
-	int numWarsWithNeighbors
+	int numWarsWithNeighbors,
+	const std::set<std::string>& majorIdeologies
 )
 {
-	if (const auto& originalFocus = loadedFocuses.find("NeighborWar"); originalFocus != loadedFocuses.end())
+	if (const auto& originalFocus = loadedFocuses.find("border_disputes"); originalFocus != loadedFocuses.end())
 	{
 		shared_ptr<HoI4Focus> newFocus = originalFocus->second.makeTargetedCopy(tag, targetNeighbors->getTag());
-		newFocus->id = "NeighborWar" + targetNeighbors->getTag() + tag;
+		newFocus->relativePositionId = "";
```

Comment:
.clear() is the "correct" way to clear a string, that way the intention is ... clear ... and I'm not questioning if this is clearing or initialization.